shared/lsif/definition-hover.test.ts

```diff
@@ -0,0 +1,95 @@
+                        { resource: resource3, range: range3 },
+                    ],
+                },
+                hover: {
+                    markdown: { text: 'foo' },
+                    range: range1,
+                },
+            })
+        )
+
+        assert.deepEqual(await definitionAndHoverForPosition(document, position, queryGraphQLFn), {
+            definition: [
+                new sourcegraph.Location(new URL('git://repo1?deadbeef1#/a.ts'), range1),
+                new sourcegraph.Location(new URL('git://repo2?deadbeef2#/b.ts'), range2),
+                new sourcegraph.Location(new URL('git://repo3?deadbeef3#/c.ts'), range3),
+            ],
+            hover: {
+                contents: {
+                    value: 'foo',
+                    kind: 'markdown',
+                },
+                range: range1,
+            },
+        })
+    })
+
+    it('should deal with empty definition payload', async () => {
+        const queryGraphQLFn = sinon.spy<QueryGraphQLFn<GenericLSIFResponse<DefinitionAndHoverResponse | null>>>(() =>
+            makeEnvelope({
+                hover: {
+                    markdown: { text: 'foo' },
+                    range: range1,
+                },
+            })
+        )
+
+        assert.deepEqual(await definitionAndHoverForPosition(document, position, queryGraphQLFn), {
+            definition: null,
+            hover: {
+                contents: {
+                    value: 'foo',
+                    kind: 'markdown',
+                },
+                range: range1,
+            },
+        })
+    })
+
+    it('should deal with empty hover payload', async () => {
+        const queryGraphQLFn = sinon.spy<QueryGraphQLFn<GenericLSIFResponse<DefinitionAndHoverResponse | null>>>(() =>
+            makeEnvelope({
+                definitions: {
+                    nodes: [
+                        { resource: resource1, range: range1 },
+                        { resource: resource2, range: range2 },
+                        { resource: resource3, range: range3 },
+                    ],
+                },
+            })
+        )
+
+        assert.deepEqual(await definitionAndHoverForPosition(document, position, queryGraphQLFn), {
+            definition: [
+                new sourcegraph.Location(new URL('git://repo1?deadbeef1#/a.ts'), range1),
+                new sourcegraph.Location(new URL('git://repo2?deadbeef2#/b.ts'), range2),
+                new sourcegraph.Location(new URL('git://repo3?deadbeef3#/c.ts'), range3),
+            ],
+            hover: null,
+        })
+    })
+
+    it('should deal with empty payload', async () => {
+        const queryGraphQLFn = sinon.spy<QueryGraphQLFn<GenericLSIFResponse<DefinitionAndHoverResponse | null>>>(() =>
+            makeEnvelope()
+        )
+
+        assert.deepEqual(await definitionAndHoverForPosition(document, position, queryGraphQLFn), null)
```

Comment:
Question here on the behavior: does this mean that when definition & hover text are not found (empty) that we display to the user a hover UI that is blank or is there someplace else in the stack where we populate a "not found" message to inform the user vs blank response?

Reply:
If there's a definition and no hover, we show buttons but no text (I think - it's possible we have canned text, or we may not display the buttons in this case either).

If there's a hover but no def, we just don't show action buttons (that I know for sure).

Reply:
Ok just wanted to make sure it was handled somewhere.